python/bucketbase/namedlock.py

```diff
@@ -0,0 +1,53 @@
+import threading
+from pathlib import Path
+
+from bucketbase.file_lock import FileLockForPath
+
+
+class NamedLockManager:
```

Comment:
Since the class is abstract, it should derive from `ABC`, and it would be cleaner to indicate in the name that it's an abstract class, like `BaseNamedLockManager` or `INamedLockManager`